packages/ses/src/module-link.js

```diff
@@ -39,51 +44,71 @@ export const link = (
   // compartment is in context: the module record may be in another
   // compartment, denoted by moduleRecord.compartment.
   // eslint-disable-next-line no-use-before-define
-  return instantiate(
-    compartmentPrivateFields,
-    moduleAnalyses,
-    moduleAliases,
-    moduleRecord,
-  );
+  return instantiate(compartmentPrivateFields, moduleAliases, moduleRecord);
 };
 
-const validateStaticModuleRecord = (staticModuleRecord, moduleAnalyses) => {
-  const isObject = typeof staticModuleRecord === 'object';
-  if (!isObject) {
-    throw new TypeError(
-      `importHook must return a StaticModuleRecord, got ${staticModuleRecord}`,
-    );
-  }
+function isPrecompiled(staticModuleRecord) {
+  return typeof staticModuleRecord.__functorSource__ === 'string';
+}
 
-  const hasAnalysis = moduleAnalyses.has(staticModuleRecord);
-  const hasImports = Array.isArray(staticModuleRecord.imports);
-  const hasExports = typeof staticModuleRecord.execute === 'function';
-
-  if (!hasAnalysis && !hasExports) {
-    if (hasImports) {
-      // In this case, the static module record has an `imports` property, but
-      // no `execute` method.
-      // This could either be a partially implemented custom static module
-      // record or created by `StaticModuleRecord` in another realm.
-      throw new TypeError(
-        'importHook must return a StaticModuleRecord constructed within the same Realm, or a custom record with both imports and an execute method',
-      );
-    } else {
-      // In this case, the static module record has no `imports` or `execute`
-      // property, so it could not have been created by the
-      // `StaticModuleRecord` from any realm.
-      // From this we infer the intent was to produce a valid custom static
-      // module record and clue accordingly.
-      throw new TypeError(
-        'importHook must return a StaticModuleRecord with both imports and an execute method',
-      );
-    }
-  }
-};
+function validatePrecompiledStaticModuleRecord(staticModuleRecord) {
+  const {
+    __fixedExportMap__,
+    __liveExportMap__,
+    __exportAlls__,
+  } = staticModuleRecord;
+  assert(
+    isObject(__fixedExportMap__),
+    `Property '__fixedExportMap__' of a precompiled module record must be an object, got ${q(
+      __fixedExportMap__,
+    )}`,
+  );
+  assert(
+    isObject(__liveExportMap__),
+    `Property '__liveExportMap__' of a precompiled module record must be an object, got ${q(
+      __liveExportMap__,
+    )}`,
+  );
+  assert(
+    isArray(__exportAlls__),
+    `Property '__exportAlls__' of a precompiled module record must be an array, got ${q(
+      __exportAlls__,
+    )}`,
+  );
+}
+
+function isThirdParty(staticModuleRecord) {
+  return typeof staticModuleRecord.execute === 'function';
+}
```

Comment:
Is this distinction still a thing? Should it still be called "third party"?

Reply:
The distinction still exists and third-party is still a good name, from the point of view of the standard if not the vagaries of the shim’s implementation.